WPFWijkagent/WPFWijkagent/MainWindow.xaml.cs

```diff
@@ -91,14 +93,51 @@ private void FillOffenceList()
         public void Pushpin_MouseDown(object sender, MouseButtonEventArgs e)
         {
             Pushpin pin = (Pushpin)sender;
+            Offence Offence = MainWindowController.RetrieveOffence(
+                pin.Location.Latitude,
+                pin.Location.Longitude);
+
+            //create a scraper without the hashtag to get all the messages for this offence and update the messages
+            Scraper SocialScraper = new Scraper(Offence);
+            SocialScraper.UpdateSocialMediaMessages();
 
-            social = new SocialMediaDialogue( 
+            //create a scraper for the witness messages and update the messages
+            Scraper WitnessScraper = new Scraper(Offence, true, Hashtag(Offence));
+            WitnessScraper.UpdateSocialMediaMessages(1);
+
+            social = new DelictDialog(pin,
                 MainWindowController.RetrieveOffence(
-                    pin.Location.Latitude, 
-                    pin.Location.Longitude));
+                    pin.Location.Latitude,
+                    pin.Location.Longitude), this);
             social.Show();
         }
 
+        /// <summary>
+        /// Create a Search hashtag fot the offence
+        /// </summary>
+        /// <param name="offence">the offence for the search hastag</param>
+        /// <returns></returns>
+        public string Hashtag(Offence offence)
+        {
+            DBContext dBContext = new DBContext();
+            SqlCommand query = new SqlCommand("SELECT Hashtag FROM Offence WHERE ID = @OffenceID");
+            query.Parameters.Add("@OffenceID", System.Data.SqlDbType.Int);
+            query.Parameters["@OffenceID"].Value = offence.ID;
+            List<object[]> rows = dBContext.ExecuteSelectQuery(query);
+
+            if (rows.Count == 1 && rows[0].GetValue(0).ToString().Length > 0)
+            {
+                string text = rows[0].GetValue(0).ToString();
+                offence.CallHashtag = text;
+                return text;
+            }
```

Comment:
Dit mag dus niet hier he..... moet in de socialmediamessagcontroller architectuur schennis!!